### PR TITLE
Improve the edge case handling within the parse tree visitor.

### LINKED ISSUE #283: panic when parsing `func{{a}}`
**Describe the bug**

cel-go will panic if the input is ``func{{a}}``

**To Reproduce**
Check which components this affects:

- [x]  parser
- [ ]  checker
- [ ]  interpreter

Sample expression and input that reproduces the issue:
```cel
func{{a}}
```

add 

```
{
		I: `func{{a}}`,
		E: ``,
},
```

to `testCases` in `parser/parser_test.go` and run `TestParse`

```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 35 [running]:
testing.tRunner.func1(0xc00014a200)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x15d05e0, 0xc00003a240)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/google/cel-go/parser.(*parser).VisitIFieldInitializerList(0xc00000e060, 0x16fcc20, 0xc00019e360, 0x4, 0xc000033601)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:315 +0x41c
github.com/google/cel-go/parser.(*parser).VisitCreateMessage(0xc00000e060, 0xc00019b890, 0xc00019b890, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:301 +0x12d
github.com/google/cel-go/parser.(*parser).VisitMemberExpr(0xc00000e060, 0xc000011900, 0x100bde5, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:231 +0x1ee
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16efa80, 0xc000011900, 0x16efa80, 0xc000011900)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:106 +0x686
github.com/google/cel-go/parser.(*parser).VisitCalc(0xc00000e060, 0xc00019b050, 0x100bde5, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:202 +0x381
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16ef580, 0xc00019b050, 0x16ef580, 0xc00019b050)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:102 +0x4d2
github.com/google/cel-go/parser.(*parser).VisitRelation(0xc00000e060, 0xc00019b020, 0x16ec940, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:184 +0x381
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16efc00, 0xc00019b020, 0x16efc00, 0xc00019b020)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:100 +0x36e
github.com/google/cel-go/parser.(*parser).VisitConditionalAnd(0xc00000e060, 0xc000168a80, 0x8, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:168 +0x7e
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16ef600, 0xc000168a80, 0x16ef600, 0xc000168a80)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:96 +0x6e2
github.com/google/cel-go/parser.(*parser).VisitConditionalOr(0xc00000e060, 0xc000168a00, 0x4, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:153 +0x7e
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16ef680, 0xc000168a00, 0x16ef680, 0xc000168a00)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:98 +0x52e
github.com/google/cel-go/parser.(*parser).VisitExpr(0xc00000e060, 0xc0000b1a40, 0x100bde5, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:141 +0x6a
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16ef880, 0xc0000b1a40, 0x16ef880, 0xc0000b1a40)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:94 +0x442
github.com/google/cel-go/parser.(*parser).VisitStart(0xc00000e060, 0xc00019adb0, 0x8, 0x100ac9b)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:136 +0x5d
github.com/google/cel-go/parser.(*parser).Visit(0xc00000e060, 0x16efd00, 0xc00019adb0, 0x16efd00, 0xc00019adb0)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:92 +0x5a6
github.com/google/cel-go/parser.(*parser).parse(0xc00000e060, 0xc00003e190, 0x9, 0xd)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:85 +0x333
github.com/google/cel-go/parser.ParseWithMacros(0x16edb40, 0xc000158000, 0x1b236e0, 0x7, 0x7, 0xc000158000, 0x4aec1712620ff)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:59 +0x237
github.com/google/cel-go/parser.Parse(...)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser.go:45
github.com/google/cel-go/parser.TestParse.func1(0xc00014a200)
	/Users/virusdefender/Desktop/workspace/cel-go/parser/parser_test.go:125 +0xb8
testing.tRunner(0xc00014a200, 0xc0000e4cd0)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
```

**Expected behavior**

the parse result is syntax error

### PR DESCRIPTION
**Pull request by TristonianJones**
Ensure that the syntax errors are handled gracefully within the parse tree visitor to avoid panics.

Closes #283

### CODE REVIEW

File: parser/parser.go [hunk 1]
```diff
@@ -307,13 +315,24 @@ func (p *parser) VisitCreateMessage(ctx *gen.CreateMessageContext) interface{} {
...
-		initID := p.helper.id(ctx.GetCols()[i])
-		value := p.Visit(ctx.GetValues()[i]).(*exprpb.Expr)
+		if i >= len(cols) {
+			// This is the result of a syntax error detected elsewhere.
+			return []*exprpb.Expr_CreateStruct_Entry{}
+		}
+		if i >= len(vals) {
+			// This is the result of a syntax error detected elsewhere.
+			return []*exprpb.Expr_CreateStruct_Entry{}
+		}
```
Comment (madi8229) [inline parser/parser.go:329]: I think these two return values are the same, should we combine them together?

File: parser/parser.go [hunk 2]
```diff
@@ -389,31 +408,28 @@ func (p *parser) VisitConstantLiteral(ctx *gen.ConstantLiteralContext) interface
...
-		key := p.Visit(ctx.GetKeys()[i]).(*exprpb.Expr)
-		value := p.Visit(ctx.GetValues()[i]).(*exprpb.Expr)
+		if i >= len(keys) {
+			// This is the result of a syntax error detected elsewhere.
+			return []*exprpb.Expr_CreateStruct_Entry{}
+		}
+		if i >= len(vals) {
+			// This is the result of a syntax error detected elsewhere.
+			return []*exprpb.Expr_CreateStruct_Entry{}
+		}
```
Comment (madi8229) [inline parser/parser.go:422]: Same as above.

File: parser/parser.go [hunk 3]
```diff
@@ -389,31 +408,28 @@ func (p *parser) VisitConstantLiteral(ctx *gen.ConstantLiteralContext) interface
 	return p.reportError(ctx, "invalid literal")
 }
 
-// Visit a parse tree produced by CELParser#exprList.
-func (p *parser) VisitExprList(ctx *gen.ExprListContext) interface{} {
```
Comment (madi8229) [inline parser/parser.go:393]: Just want to confirm that there is no other function call this function.

**Review by madi8229 — APPROVED** [review]

File: parser/parser.go (hunk 3 above)
Comment (TristonianJones) [reply parser/parser.go:393]: Yep, it looks like this was dead code since it has no callers.

**Pull request merged**